flink-runtime/src/main/java/org/apache/flink/runtime/failure/FailureEnricherUtils.java

```diff
@@ -106,23 +108,34 @@ static Collection<FailureEnricher> getFailureEnrichers(
             }
         }
 
+        if (!includedEnrichers.values().stream().allMatch(Boolean::booleanValue)) {
```

Comment:
i think here can use `includedEnrichers.removeAll(failureEnrichers)` directly.
It is not necessary to modify the return type of `getIncludedFailureEnrichers`. The boolean in the return type of this `getIncludedFailureEnrichers` method is not easy to explain.

Reply:
how do you think?

Reply:
I agree, I went back and forth between changing the method signature but this simplifies things a lot.
Updated, and thanks ;)